test/integration/model/count.test.js

```diff
@@ -133,5 +133,47 @@ describe(Support.getTestDialectTeaser('Model'), function() {
       });
     });
 
+    it('should be able to use where clause on included models', function() {
+      const queryObject = {
+        col: 'username',
+        include: [this.Project],
+        where: {}
+      };
+      queryObject.where['$Projects.name$'] = 'project1';
+      return this.User.count(queryObject).then((count) => {
+        expect(parseInt(count)).to.be.eql(1);
+        const queryObject = {
+          col: 'username',
+          include: [this.Project],
+          where: {}
+        };
+        queryObject.where['$Projects.name$'] = 'project2';
+        return this.User.count(queryObject);
+      }).then((count) => {
+        expect(parseInt(count)).to.be.eql(0);
+      });
+    });
+
+    it('should have the same count result as findAndCount', function() {
+      let result;
+      const queryObject = {
+        include: [this.Project],
+        logging: function (sql) {
```

Comment:
Hmm, that feels a bit dirty 

You could also write this as a unit test, where you verify that the arguments to find and count are correct